app/src/components/GameRoutes.tsx

```diff
@@ -158,12 +159,21 @@ function CurrentGameProvider(props: {
   joinCode: string
   children: React.ReactNode
 }) {
-  const { data, loading } = useCurrentGameSubscription({
+  const { data, loading, subscribeToMore } = useCurrentGameQuery({
     variables: {
       joinCode: props.joinCode,
     },
   })
 
+  React.useEffect(() => {
+    return subscribeToMore({
+      variables: {
+        joinCode: props.joinCode,
+      },
+      document: CurrentGameSubDocument,
+    })
+  }, [data?.games[0]])
```

Comment:
subscribe this way, then can use optimistic response instead of local states.

related to: https://github.com/apollographql/apollo-client/issues/5267

Reply:
some interesting behavior here - some flashes on refresh, and two web sockets get opened.

Reply:
Yeah, I ran into this same thing recently and landed on the same GitHub issue / solution; it worked well.

You're resubscribing a bunch in the current state and missing an `updateQuery` which might help:

```suggestion
  React.useEffect(() => {
    return subscribeToMore({
      variables: {
        joinCode: props.joinCode,
      },
      document: CurrentGameSubDocument,
      updateQuery: (data, { subscriptionData }) => {
        return subscriptionData.data || data
      },
    })
  }, [props.joinCode])
```

The subscription and query are the same in this context, so `updateQuery` is dead simple, but it gives you the flexibility to deviate from the query (i.e. only make some things realtime, subscribe to a handful of smaller subscriptions, etc).

Reply:
Yea - I was meaning to check if updateQuery defaults to exactly that if not passed in - but will add it explicitly to see the effect.

I also thought direction of flexibility was 👍 e.g... we are subscribing for everything once game is started. But practically... the most settings, players and cards cannot change after the game has started, so based on the game state we could actually run different subscriptions (so long as we close out the previous gracefully).

Reply:
> I was meaning to check if updateQuery defaults to exactly that if not passed in

It [_seems_ like](https://github.com/apollographql/apollo-client/blob/v2.6.8/packages/apollo-client/src/core/ObservableQuery.ts#L408-L416) it doesn't on glance, but I'm not sure how it would have been working without it now if that's the case 🤔